File: Java/src/String to Integer (atoi).java
public class Solution {
    public int atoi(String str) {
        boolean sign = false;
        int isNeg = 1;
        double ret = 0;
        str = str.trim();

        for (int i = 0; i<str.length(); i++){
            char c = str.charAt(i);
            if ((c == '-' || c == '+') && !sign ) { 
                sign = true;
                if (c== '-'){
                    isNeg = -1;
                }
            }
            else if (c >= '0' && c <='9'){
                ret = ret * 10 + Character.getNumericValue(c);
            }
            else{
                break;
            }
        }
        int intRet = (int)(ret*isNeg);
        if (intRet > Integer.MAX_VALUE){
            return Integer.MAX_VALUE;
        }
        if (intRet < Integer.MIN_VALUE){
            return Integer.MIN_VALUE;
        }

        return intRet;
    }
}

File: Python/String to Integer (atoi).py

class Solution:
    # @return an integer
    def atoi(self, str):
        intmax = 2147483647
        intmin = -2147483648
        isPos = True
        ret = 0
        str = str.strip()
        if not str:
            return 0
        if str[0] == "-":
            isPos = False
        elif str[0] == "+":
            isPos = True
        elif 48 <= ord(str[0]) <=57 :
            ret = ord(str[0])-48
        else:
            return 0
        
        str = str[1:]
        for s in str:
            l = ord(s)-48
            if 0 <= l <= 9:
                ret = ret*10 + l
            else:
                if not isPos:
                    ret = -ret
                if ret > intmax:
                    return intmax
                if ret < intmin:
                    return intmin
                return ret

        if not isPos:
            ret = -ret
        if ret > intmax:
            return intmax

        if ret < intmin:
            return intmin
            
        return ret
        